library/management/chat/src/main/java/org/quiltmc/qsl/chat/api/types/AbstractChatMessage.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2023 QuiltMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.quiltmc.qsl.chat.api.types;
+
+import net.minecraft.entity.player.PlayerEntity;
+import org.jetbrains.annotations.NotNull;
+import org.quiltmc.qsl.chat.api.QuiltMessageType;
+
+import java.util.EnumSet;
+
+/**
+ * An abstract message, extended for the various types of messages.
+ *
+ * @param <S> The type of the serialized form of this class, usually a Packet<?> but not required
+ */
+public abstract class AbstractChatMessage<S> {
+	protected final @NotNull PlayerEntity player;
+	protected final boolean isOnClientSide;
```

Comment:
```suggestion
	protected final boolean isClient;
```
This should go for the rest of the same name.

Reply:
Oh, do note that `is` violates QM's conventions :P

Reply:
https://github.com/QuiltMC/quilt-mappings/blob/d7fa31ee0a3f7620bbaf164e4e0b1058bf77bbec/mappings/net/minecraft/world/World.mapping#L25

Reply:
The point still stands:
https://github.com/QuiltMC/quilt-standard-libraries/blob/1.19.3/CONTRIBUTING.md#general

Reply:
> avoiding the `is` and `has` prefixes when possible
Its not possible to be `cliented`, so its fine in this case.

Reply:
I could do `clientSided` though

Reply:
oh well, fair enough

Reply:
im mostly trying to have it match what QM currently uses for clarity

Reply:
Makes sense